pkg/cloud/aliyunprovider.go

```diff
@@ -842,3 +919,52 @@ func generateSlimK8sNodeFromV1Node(node *v1.Node) *SlimK8sNode {
 
 	return NewSlimK8sNode(instanceType, regionID, priceUnit, memorySizeInKiB, osType, providerID, instanceFamily, IsIoOptimized)
 }
+
+// generateSlimK8sDiskFromV1PV function generates SlimK8sDisk from v1.PersistentVolume and DescribeDisk API(If required) of alibaba
+// to generate slim disk type that can be used to fetch pricing information.
+func generateSlimK8sDiskFromV1PV(pv *v1.PersistentVolume, regionID string) *SlimK8sDisk {
+	//Regular expression to get the GiB storage size for the API call
+	sizeRegEx := regexp.MustCompile("(.*?)Gi")
```

Comment:
MustCompile is a little spicy -- would you please write a really simple unit test that calls this function so that we can know for certain that the regex compiles at compile/test time?

I'd also recommend pulling the compilation of the regex into a top-level var (outside of a function scope) so that we aren't recompiling this regex every time we call this function.

Reply:
Definitely need to pull this out into a var outside the func

Reply:
Adopted it, Thanks!